### ISSUE TITLE
Suggestion: Pixellation and Wave Filter

### ISSUE THREAD

**Issue by SolarLune** [GoranM/bdx#412]
Hey, there. 

So, a built-in Pixellation filter might be nice to have. They're frequently used for games for transitions (think RPGs like Pokemon) or possibly blurring out background areas or achieving lo-fi effects.

A Wave filter would be nice to have. It would just slide rows or columns of the screen around as you want it. It could be used for hot waving air, or transitions as well (think the Dragon Warrior Monsters Gameboy Color games, IIRC). I'd also see about adding a value to make the wave decrease with screen height or width.

**Comment by GoranM** [comment]
Ok.

**Comment by SolarLune** [comment]
I ended up doing a fairly standard (?) wave filter (no making the wave decrease depending on the screen), and a pixellation filter where you set your targeted resolution (so the pixels gets bigger or smaller depending on the resolution of the window). 

I also added a new set of uniforms, the screenWidth and screenHeight uniforms, which correspond to the screen size (needed for the pixellate filter, and also generally useful to have for 2D screen filters). 

Also, I fixed it so that the frameBuffer is cleared before its written to from the temp buffer. By writing directly over the frameBuffer before, the unfiltered scene was seen through the filtered result.
